### PR TITLE
Modernize codebase (v2.x) and QA

### PR DESCRIPTION
**Pull request by gmazzap**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Maintenance.

**What is the current behavior?** (You can also link to an open issue here)

N/A

**What is the new behavior (if this is a feature change)?**

No new behavior.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No.

**Other information**:

- Require PHP 7.4+ and WP 5.5+ (so we can rely on `wp_get_environment_type()`)
- Update all dependencies' requirements
- Update code to be compliant with the latest coding standards and Psalm
- Deprecate "Spaces" hosting

### CODE REVIEW

File: src/EnvConfig.php [hunk 1]
```diff
@@ -18,63 +18,29 @@ class EnvConfig implements SiteConfig
...
-        'stage' => self::STAGING,
-        'preprod' => self::STAGING,
-        'pre-prod' => self::STAGING,
-        'pre-production' => self::STAGING,
-        'test' => self::STAGING,
-        'uat' => self::STAGING,
-        'production' => self::PRODUCTION,
-        'prod' => self::PRODUCTION,
-        'live' =>  self::PRODUCTION,
-    ];
```
Comment (Chrico) [inline src/EnvConfig.php:36]: This was replaced by using `wp_get_environmen_type()`. Will this affect current systems which are using the non-defined WordPress env types (local, development, staging, production)?

**Review by Chrico — COMMENTED** [review]
Looks good to me so far. Only have 1 question regarding the replacment of our ENV_ALIAS with the `wp_get_enviornment_type()` core function.

File: src/EnvConfig.php (hunk 1 above)
Comment (gmazzap) [reply src/EnvConfig.php:36]: The replaced code was still reducing the env names to the one supported by WP. Because after being obtained, the [environment was "normalized"](https://github.com/inpsyde/wp-app-container/blob/c2a09ac52fbafb4a6083998765947b87b8687725/src/EnvConfig.php#L179) to [one of the class constants](https://github.com/inpsyde/wp-app-container/blob/master/src/EnvConfig.php#L16-L19) which are exactly the values returned by `wp_get_environmen_type()`.

Moreover, if you were using a WP version that has `wp_get_environment_type()` available (and considering that's 5.5+ chances are you are) [**the class was _already_ using `wp_get_environment_type()`**](https://github.com/inpsyde/wp-app-container/blob/c2a09ac52fbafb4a6083998765947b87b8687725/src/EnvConfig.php#L169-L170).

In short, this change is just removing a good amount of complexity that was there just to account for WP < 5.5, which we don't want to care anymore...
